src/transformers/models/moonshine/modeling_moonshine.py

```diff
@@ -260,6 +258,8 @@ def forward(
                 key_states, value_states = past_key_value.update(
                     key_states, value_states, self.layer_idx, {"cache_position": cache_position}
                 )
+                # set flag that curr layer for cross-attn is already updated so we can re-use in subsequent calls
+                past_key_value.is_updated[self.layer_idx] = True
```

Comment:
took this pattern from T5 and modified everywhere: it is much clearer than the one originally in whisper